README.rst
Python Client for Recommendations AI API
========================================

|preview| |pypi| |versions|

`Recommendations AI API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |preview| image:: https://img.shields.io/badge/support-preview-orange.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-recommendations-ai.svg
   :target: https://pypi.org/project/google-cloud-recommendations-ai/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-recommendations-ai.svg
   :target: https://pypi.org/project/google-cloud-recommendations-ai/
.. _Recommendations AI API: https://cloud.google.com/recommendations-ai/
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/recommendationengine/latest
.. _Product Documentation:  https://cloud.google.com/recommendations-ai/

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Recommendations AI API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Recommendations AI API.:  https://cloud.google.com/recommendations-ai/
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-recommendations-ai


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-recommendations-ai

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Recommendations AI API
   to see other available methods on the client.
-  Read the `Recommendations AI API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Recommendations AI API Product documentation:  https://cloud.google.com/recommendations-ai/
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

### [0.3.4](https://www.github.com/googleapis/python-recommendations-ai/compare/v0.3.3...v0.3.4) (2021-10-05)


### Bug Fixes

* improper types in pagers generation ([2d73287](https://www.github.com/googleapis/python-recommendations-ai/commit/2d732875566d49e9eed8702a40120c427222d529))

### [0.3.3](https://www.github.com/googleapis/python-recommendations-ai/compare/v0.3.2...v0.3.3) (2021-09-24)


### Bug Fixes

* add 'dict' annotation type to 'request' ([ff8c418](https://www.github.com/googleapis/python-recommendations-ai/commit/ff8c4189e70770b83b84eb0d6cf886104d62d36a))

### [0.3.2](https://www.github.com/googleapis/python-recommendations-ai/compare/v0.3.1...v0.3.2) (2021-07-28)


### Bug Fixes

* enable self signed jwt for grpc ([#72](https://www.github.com/googleapis/python-recommendations-ai/issues/72)) ([2244581](https://www.github.com/googleapis/python-recommendations-ai/commit/22445819af2b11d8cc1d62d7e7b5265ffd950cdd))


### Documentation

* add Samples section to CONTRIBUTING.rst ([#68](https://www.github.com/googleapis/python-recommendations-ai/issues/68)) ([190c5b0](https://www.github.com/googleapis/python-recommendations-ai/commit/190c5b04a6d01b006ebb81fe95142a760087c574))


### Miscellaneous Chores

* release as 0.3.2 ([#73](https://www.github.com/googleapis/python-recommendations-ai/issues/73)) ([67460b7](https://www.github.com/googleapis/python-recommendations-ai/commit/67460b7c78ebbdbc0b3fd45623a8ff325f1bb86e))

### [0.3.1](https://www.github.com/googleapis/python-recommendations-ai/compare/v0.3.0...v0.3.1) (2021-07-20)


### Bug Fixes

* **deps:** pin 'google-{api,cloud}-core', 'google-auth' to allow 2.x versions ([#67](https://www.github.com/googleapis/python-recommendations-ai/issues/67)) ([cd12659](https://www.github.com/googleapis/python-recommendations-ai/commit/cd12659af98033da94724ba28f5f942f5b43e732))

## [0.3.0](https://www.github.com/googleapis/python-recommendations-ai/compare/v0.2.2...v0.3.0) (2021-07-01)


### Features

* add always_use_jwt_access ([#60](https://www.github.com/googleapis/python-recommendations-ai/issues/60)) ([354fc1c](https://www.github.com/googleapis/python-recommendations-ai/commit/354fc1cef059a69ab2b1310858f546a66db6fe5a))


### Bug Fixes

* disable always_use_jwt_access ([#64](https://www.github.com/googleapis/python-recommendations-ai/issues/64)) ([3263c2e](https://www.github.com/googleapis/python-recommendations-ai/commit/3263c2ee0eb44e283de7356dbf9410d8620a8727))


### Documentation

* omit mention of Python 2.7 in 'CONTRIBUTING.rst' ([#1127](https://www.github.com/googleapis/python-recommendations-ai/issues/1127)) ([#55](https://www.github.com/googleapis/python-recommendations-ai/issues/55)) ([47faef8](https://www.github.com/googleapis/python-recommendations-ai/commit/47faef890d8356ce60da06925b92a50f23a34e20)), closes [#1126](https://www.github.com/googleapis/python-recommendations-ai/issues/1126)

### [0.2.2](https://www.github.com/googleapis/python-recommendations-ai/compare/v0.2.1...v0.2.2) (2021-06-16)


### Bug Fixes

* exclude docs and tests from package ([#52](https://www.github.com/googleapis/python-recommendations-ai/issues/52)) ([5d4926d](https://www.github.com/googleapis/python-recommendations-ai/commit/5d4926d7220f924ba296098e102d8c64b010ad36))

### [0.2.1](https://www.github.com/googleapis/python-recommendations-ai/compare/v0.2.0...v0.2.1) (2021-05-28)


### Bug Fixes

* **deps:** add packaging requirement ([#46](https://www.github.com/googleapis/python-recommendations-ai/issues/46)) ([d2eb28f](https://www.github.com/googleapis/python-recommendations-ai/commit/d2eb28fdc016de1b9f82e46f36b8bf8c8bb7edbd))

## [0.2.0](https://www.github.com/googleapis/python-recommendations-ai/compare/v0.1.0...v0.2.0) (2021-03-29)


### Features

* add async clients ([f8e3d0f](https://www.github.com/googleapis/python-recommendations-ai/commit/f8e3d0f8d12921b926c24cf33e10c4d4390164bb))
* add common resource helper methods ([f8e3d0f](https://www.github.com/googleapis/python-recommendations-ai/commit/f8e3d0f8d12921b926c24cf33e10c4d4390164bb))


### Bug Fixes

* BREAKING rename `PriceRange.min` to `PriceRange.min_`, `PriceRange.max` to `PriceRange.max_` ([f8e3d0f](https://www.github.com/googleapis/python-recommendations-ai/commit/f8e3d0f8d12921b926c24cf33e10c4d4390164bb))
* fix bug with enums closes [#14](https://www.github.com/googleapis/python-recommendations-ai/issues/14), [#20](https://www.github.com/googleapis/python-recommendations-ai/issues/20) ([f8e3d0f](https://www.github.com/googleapis/python-recommendations-ai/commit/f8e3d0f8d12921b926c24cf33e10c4d4390164bb))

## 0.1.0 (2020-03-13)


### Features

* generate v1beta1 ([90a06d1](https://www.github.com/googleapis/python-recommendations-ai/commit/90a06d1edb5b72403c6e039f99c90d1bdeef3337))


owl-bot-staging/v1beta1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")


nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
]

@nox.session(python=['3.6', '3.7', '3.8', '3.9'])
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/recommendationengine_v1beta1/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python='3.7')
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=['3.6', '3.7'])
def mypy(session):
    """Run the type checker."""
    session.install('mypy', 'types-pkg_resources')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python='3.6')
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
